src/tm-threads.c

```diff
@@ -1713,16 +1766,32 @@ TmEcode TmThreadSpawn(ThreadVars *tv)
         return TM_ECODE_FAILED;
     }
 
-    /* Initialize and set thread detached attribute */
-    pthread_attr_init(&attr);
+#ifdef HAVE_DPDK
+    /* only DPDK worker threads are launched using rte_eal_remote_launch */
+    if (tv->lcore_id != LCORE_ID_ANY) {
+        struct dpdk_callback cb = { .func = tv->tm_func, .tv = tv };
 
-    pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_JOINABLE);
+        int rc = rte_eal_remote_launch(&TmThreadSpawnDpdkThread, (void *)&cb, tv->lcore_id);
+        if (rc != 0) {
+            SCLogError(SC_ERR_DPDK_INIT, "Err (%" PRIu32 ") Can not launch function on lcore", rc);
+            return TM_ECODE_FAILED;
+        }
 
-    int rc = pthread_create(&tv->t, &attr, tv->tm_func, (void *)tv);
-    if (rc) {
-        printf("ERROR; return code from pthread_create() is %" PRId32 "\n", rc);
-        return TM_ECODE_FAILED;
+    } else {
+#endif /* HAVE_DPDK */
+        /* Initialize and set thread detached attribute */
```

Comment:
you mention in comment the thread is detached, should not the attribute be set as detached?

Reply:
threads are now solely pthreads, so this is not relevant anymore.